stdlib/public/libexec/swift-backtrace/TargetLinux.swift

```diff
@@ -0,0 +1,232 @@
+    switch signal {
+      case UInt64(SIGQUIT): return "SIGQUIT"
+      case UInt64(SIGABRT): return "SIGABRT"
+      case UInt64(SIGBUS):  return "SIGBUS"
+      case UInt64(SIGFPE): return "SIGFPE"
+      case UInt64(SIGILL): return "SIGILL"
+      case UInt64(SIGSEGV): return "SIGSEGV"
+      case UInt64(SIGTRAP): return "SIGTRAP"
+      default: return "\(signal)"
+    }
+  }
+
+  var signalDescription: String {
+    switch signal {
+      case UInt64(SIGQUIT): return "Terminated"
+      case UInt64(SIGABRT): return "Aborted"
+      case UInt64(SIGBUS): return "Bus error"
+      case UInt64(SIGFPE): return "Floating point exception"
+      case UInt64(SIGILL): return "Illegal instruction"
+      case UInt64(SIGSEGV): return "Bad pointer dereference"
+      case UInt64(SIGTRAP): return "System trap"
+      default:
+        return "Signal \(signal)"
+    }
+  }
+
+  var reader: MemserverMemoryReader
+
+  // Get the name of a process
+  private static func getProcessName(pid: pid_t) -> String {
+    let path = "/proc/\(pid)/comm"
+    guard let name = readString(from: path) else {
+      return ""
+    }
+    return String(stripWhitespace(name))
+  }
+
+  /// Get the name of a thread
+  private func getThreadName(tid: Int64) -> String {
+    let path = "/proc/\(pid)/task/\(tid)/comm"
+    guard let name = readString(from: path) else {
+      return ""
+    }
+    let trimmed = String(stripWhitespace(name))
+
+    // Allow the main thread to use the process' name, but other
+    // threads will have an empty name unless they've set the name
+    // explicitly
+    if trimmed == self.name && pid != tid {
+      return ""
+    }
+    return trimmed
+  }
+
+  init(crashInfoAddr: UInt64, limit: Int?, top: Int, cache: Bool) {
+    // fd #4 is reserved for the memory server
+    let memserverFd: CInt = 4
+
+    pid = getppid()
+    reader = MemserverMemoryReader(fd: memserverFd)
+    name = Self.getProcessName(pid: pid)
+
+    let crashInfo: CrashInfo
+    do {
+      crashInfo = try reader.fetch(from: crashInfoAddr, as: CrashInfo.self)
+    } catch {
+      print("swift-backtrace: unable to fetch crash info.")
+      exit(1)
+    }
+
+    crashingThread = TargetThread.ThreadID(crashInfo.crashing_thread)
+    signal = crashInfo.signal
+    faultAddress = crashInfo.fault_address
+
+    images = Backtrace.captureImages(using: reader,
+                                     forProcess: Int(pid))
+
+    do {
+      try fetchThreads(threadListHead: Address(crashInfo.thread_list),
+                       limit: limit, top: top, cache: cache)
+    } catch {
+      print("swift-backtrace: failed to fetch thread information: \(error)")
+      exit(1)
+    }
+  }
+
+  /// Fetch information about all of the process's threads; the crash_info
+  /// structure contains a linked list of thread ucontexts, which may not
+  /// include every thread.  In particular, if a thread was stuck in an
+  /// uninterruptible wait, we won't have a ucontext for it.
```

Comment:
Could more context be added for each parameter in this doc comment?

Reply:
`limit` and `top` are the same as elsewhere, and I think the list head argument is obvious. I'm not inclined to update the comment here.